harvester_e2e_tests/apis/test_networks.py

```diff
@@ -99,3 +100,31 @@ def test_create_network_using_terraform(request, admin_session,
                                         harvester_api_endpoints,
                                         network_using_terraform):
     pass
+
+
+@pytest.mark.terraform
+def test_cluster_network_using_terraform(request, admin_session,
+                                         harvester_api_endpoints):
+
+    network_data = utils.create_clusternetworks_terraform(
+                       request,
+                       admin_session,
+                       harvester_api_endpoints,
+                       'resource_clusternetworks',
+                       'eth0')
+
+    time.sleep(30)
+    assert 'eth0' in network_data['config']['defaultPhysicalNIC']
+    if not request.config.getoption('--do-not-cleanup'):
+        utils.destroy_resource(request, admin_session, 'dir-only')
+
+    # undo cluster network update back to original
+    network_data = utils.create_clusternetworks_terraform(
+                        request, admin_session,
+                        harvester_api_endpoints,
+                        'resource_clusternetworks',
+                        'vlan')
```

Comment:
shouldn't this come from the `--vlan-nic` config?

Reply:
after cluster_network testing done. Undo cluster_network. Instead of hardcording vlan - I have get the existing vlan-nic value and update it back

Reply:
after sync up with @guangyee . We dont need this undo cluster network. 
Also added, enable vlan for terraform and also covered testcase for already existing public network.